x/mongo/driver/topology/topology.go

```diff
@@ -564,14 +564,10 @@ func (t *Topology) selectServerFromDescription(desc description.Topology,
 			allowedIndexes = append(allowedIndexes, i)
 		}
 	}
-	var allowed []description.Server
-	if len(allowedIndexes) == len(desc.Servers) {
-		allowed = desc.Servers
-	} else {
-		allowed = make([]description.Server, len(allowedIndexes))
-		for i, idx := range allowedIndexes {
-			allowed[i] = desc.Servers[idx]
-		}
+
+	allowed := make([]description.Server, len(allowedIndexes))
+	for i, idx := range allowedIndexes {
+		allowed[i] = desc.Servers[idx]
```

Comment:
Suggest reverting this file to before https://github.com/mongodb/mongo-go-driver/commit/a2a01cb2b8bc3ff7f09280e88acb640d119c63af. The `allowedIndexes` do not serve much purpose now.

Reply:
@kevinAlbs Allocating the slice size is still an important optimization.

Current design:

```
❯ go test -bench=BenchmarkSelectServerFromDescription
goos: darwin
goarch: arm64
pkg: go.mongodb.org/mongo-driver/x/mongo/driver/topology
BenchmarkSelectServerFromDescription/AllFit-10             98368             12378 ns/op           41856 B/op          2 allocs/op
BenchmarkSelectServerFromDescription/AllButOneFit-10       96394             12252 ns/op           41856 B/op          2 allocs/op
BenchmarkSelectServerFromDescription/HalfFit-10           190622              6118 ns/op           21376 B/op          2 allocs/op
BenchmarkSelectServerFromDescription/OneFit-10           2475796               499.3 ns/op          1312 B/op          2 allocs/op
PASS
ok      go.mongodb.org/mongo-driver/x/mongo/driver/topology     26.196s
```

Reverting:
```
❯ go test -bench=BenchmarkSelectServerFromDescription
goos: darwin
goarch: arm64
pkg: go.mongodb.org/mongo-driver/x/mongo/driver/topology
BenchmarkSelectServerFromDescription/AllFit-10             34582             34127 ns/op          111265 B/op          8 allocs/op
BenchmarkSelectServerFromDescription/AllButOneFit-10       34394             33892 ns/op          111265 B/op          8 allocs/op
BenchmarkSelectServerFromDescription/HalfFit-10            97132             15648 ns/op           53920 B/op          7 allocs/op
BenchmarkSelectServerFromDescription/OneFit-10           4002678               306.6 ns/op           416 B/op          1 allocs/op
PASS
ok      go.mongodb.org/mongo-driver/x/mongo/driver/topology     29.761s
```